include/seqan3/range/container/constexpr_vector.hpp

```diff
@@ -0,0 +1,918 @@
+    constexpr constexpr_vector(other_value_type... args) noexcept :
+        data_{args...}, sz{sizeof...(other_value_type)}
+    {
+        static_assert(sizeof...(other_value_type) < capacity_, "Value list must not exceed the capacity.");
+    }
+
+    /*!\brief Construct with `n` times `value`.
+     * \param[in] n     Number of elements.
+     * \param[in] value The initial value to be assigned.
+     *
+     * ### Complexity
+     *
+     * Linear in `n`.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr constexpr_vector(size_type n, value_type value) noexcept : constexpr_vector{}
+    {
+        assign(n, value);
+    }
+
+    /*!\brief Construct from two iterators.
+     * \tparam begin_iterator_type Must model std::ForwardIterator and `value_type` must be constructible from
+     *                             the reference type of begin_iterator_type.
+     * \tparam   end_iterator_type Must satisfy std::Sentinel.
+     * \param[in]         begin_it Begin of range to construct/assign from.
+     * \param[in]           end_it End of range to construct/assign from.
+     *
+     * ### Complexity
+     *
+     * Linear in the distance between `begin_it` and `end_it`.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    template <std::ForwardIterator begin_iterator_type, std::Sentinel<begin_iterator_type> end_iterator_type>
+    //!\cond
+        requires std::Constructible<value_type, /*ranges::iter_reference_t*/reference_t<begin_iterator_type>>
+    //!\endcond
+    constexpr constexpr_vector(begin_iterator_type begin_it, end_iterator_type end_it) noexcept : constexpr_vector{}
+    {
+        assign(begin_it, end_it);
+    }
+
+    /*!\brief Construct from a different range.
+     * \tparam other_range_t The type of range to be inserted; must satisfy std::ranges::InputRange and `value_type`
+     *                       must be constructible from reference_t<other_range_t>.
+     * \param[in]      range The sequences to construct/assign from.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of `range`.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    template <std::ranges::InputRange other_range_t>
+    //!\cond
+        requires !std::is_same_v<remove_cvref_t<other_range_t>, constexpr_vector>
+                 /*ICE: && std::Constructible<value_type, reference_t<other_range_t>>*/
+    //!\endcond
+    constexpr constexpr_vector(other_range_t && range) noexcept :
+        constexpr_vector{std::ranges::begin(range), std::ranges::end(range)}
+    {}
+    //!\}
+
+    /*!\brief Assign from `std::initializer_list`.
+     * \param[in] ilist A `std::initializer_list` of value_type.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of `ilist`.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr constexpr_vector & operator=(std::initializer_list<value_type> ilist) noexcept
```

Comment:
Regarding adding `noexcept` everywhere. I was maybe a little quick. This depends, of course, on the value_type being no no_throw_copy_constructible / no_throw_move_constructible. I would maybe add a static member variable `is_noexcept = std::no_throw_copy_constructible_v<value_type>` and then change the `noexcept` on the constructors, assignment, insert... to tbe `noexcept(is_noexcept)`. The documentation would also need to be adapted.

Reply:
check after my next push if i have all the places correct